src/ui/PipelineContext/PipelineContext.ts

```diff
@@ -110,12 +146,38 @@ export class PipelineContext extends Component {
     });
   }
 
-  // We need to modify the data to escape special salesforce characters. eg: {! }
+  private tryParseContextFromString(contextAsString: string): Context {
+    if (_.isEmpty(contextAsString)) {
+      return {};
+    }
+    try {
+      // Context could be HTML encoded (eg: Coveo for Salesforce)
+      return JSON.parse(Utils.decodeHTMLEntities(contextAsString));
+    } catch (e) {
+      try {
+        return JSON.parse(contextAsString);
+      } catch (e) {
+        this.logger.error(`Error while trying to parse context from the PipelineContext component`, e);
+        return null;
+      }
+    }
+  }
+
   private getModifiedData(value: string) {
+    /* We need to modify the data to escape special salesforce characters. eg: {! }
+     If we find the matching value in the global Coveo.context variable, we return that one instead of the one present locally.
+     So, concretely, the component could contain : 
+     {
+       "productName" : "{! productValueFromSalesforce }"
+     }
+
+     This means that in those case, we would try to access Coveo.context.productValueFromSalesforce (which would in theory be a "real" product value from salesforce, and not a placeholder/variable)
+    */
     return value.replace(/\{\!([^\}]+)\}/g, (all: string, contextKey: string) => {
-      if (Coveo.context != null && contextKey in Coveo.context) {
-        return Coveo.context[contextKey];
-      } else if (contextKey == PipelineContext.CURRENT_URL) {
+      const trimmedKey = contextKey.trim();
+      if (Coveo.context != null && trimmedKey in Coveo.context) {
```

Comment:
Rather than checking if `Coveo.context` is not not defined, we could check if it is defined.